server/app/src/mocap_server.py

```diff
@@ -642,56 +609,25 @@ def updateStatusFile(self):
                 'protected/server_status/'+self.serverId, statusStr)
             print('Uploaded updated status file: \n' + statusStr)
 
-    def cleanUpOtherDeadServersForever(self):
-        """
-        This will spin, pinging anyone else whose server status files are still online at intervals, 
-        and cleaning up servers that haven't responded for more than a configured timeout.
-        """
-        pingIntervalSeconds = 30
-        deathIntervalSeconds = 60
+    def onPubSubStatusReceived(self, topic: str, payload: bytes):
+        print(f'Received PubSub status update on server {self.serverId}')
+        self.index.pubSub.alive = True
+
+    def checkPubSubStatusForever(self):
 
         while True:
-            statusFiles: Dict[str, FileMetadata] = self.index.getImmediateChildren(
-                "protected/server_status/")
+            # First, assume that PubSub is down.
+            self.index.pubSub.alive = False
 
-            # Send out pings
-            for k in statusFiles:
-                if k == self.serverId:
-                    pass
-                else:
-                    # Send a ping, which will get an asynchronous response which will eventually update self.lastSeenPong[k]
-                    try:
-                        self.index.pubSub.sendMessage('/PING/'+k, {})
-                    except e:
-                        print('Failed to send ping to ' +
-                              k+': '+str(e), flush=True)
+            # Send a status update message and wait a few seconds.
+            self.index.pubSub.publish('/PUBSUB_STATUS/'+self.serverId, {})
```

Comment:
This is the other place to ensure topic length, perhaps by using a shorter hash ID